include/zephyr/net/socket_offload.h

```diff
@@ -41,6 +41,39 @@ struct socket_dns_offload {
  */
 void socket_offload_dns_register(const struct socket_dns_offload *ops);
 
+/**
+ * @brief Deregister an offloaded socket DNS API interface.
+ *
+ * @param ops A pointer to the offloaded socket DNS API interface.
+ *
+ * @retval 0 On success
+ * @retval -EINVAL Offloaded DNS API was not regsitered.
+ */
+int socket_offload_dns_deregister(const struct socket_dns_offload *ops);
```

Comment:
Register/deregister API functions does make it look like multiple instances could be registered simultaneously. Could this even be supported in the future maybe?

Reply:
I think it could, not the case for now. And I'm not sure how should `getaddrinfo()` behave in such case, call one offloaded implementation after another until one succeeds? The more I think about the `getaddrinfo()` offloading I'm getting into impression offloading that function at such a high level wasn't the best idea.